ccm_web/client/src/pages/GradebookCanvas.tsx

```diff
@@ -0,0 +1,395 @@
+  }
+
+  const handleParseSuccess = (grades: GradebookRecord[]): void => {
+    setPageState({ state: GradebookCanvasPageState.Confirm, grades: grades })
+    setCSVtoDownload(grades)
+  }
+
+  const handleParseComplete = (results: ParseResult<GradebookRecord>): void => {
+    const data = results.data.slice(1) // The first row is possible scores
+
+    if (data[0]['Final Grade'] === undefined) {
+      handleNoLetterGradesError()
+      return
+    }
+
+    let invalidations: GradebookRowInvalidation[] = []
+
+    const gradeMismatchValidator = new CurrentAndFinalGradeMatchGradebookValidator()
+
+    data.forEach(record => {
+      invalidations = invalidations.concat(gradeMismatchValidator.validate(record, data.indexOf(record) + 1))
+    })
+
+    if (invalidations.length > 0) {
+      handleRowLevelInvalidationError([<div key='0'>There are blank cells in the gradebook. Please enter 0 or EX (for excused) for any blank cells in the gradebook and export a new CSV file.</div>], invalidations)
+    } else {
+      handleParseSuccess(data)
+    }
+  }
+
+  const renderUploadHeader = (): JSX.Element => {
+    return <span>
+      <Typography variant='h6'>Upload your CSV File</Typography>
+      <Typography>The CSV file will be formatted by trimming out nonessential columns.</Typography>
+      <br/>
+      <Typography><strong>Requirement needed:</strong> <Link href='#'>Grading Scheme in settings</Link> needs to be check marked for letter grade to appear in the CSV file.</Typography>
+    </span>
+  }
+
+  const renderFileUpload = (): JSX.Element => {
+    return <span>
+      <Grid container>
+        <Grid item xs={12}>
+          <FileUpload onUploadComplete={uploadComplete}></FileUpload>
+        </Grid>
+      </Grid>
+    </span>
+  }
+
+  const renderUpload = (): JSX.Element => {
+    return <span>
+      {renderUploadHeader()}
+      <br/>
+      {renderFileUpload()}
+    </span>
+  }
+
+  const renderUploadAgainButton = (): JSX.Element => {
+    return <Button color='primary' component="span" onClick={() => resetPageState()}>Upload again</Button>
+  }
+  const renderRowLevelErrors = (invalidations: GradebookRowInvalidation[]): JSX.Element => {
+    return (
+      <div>
+        {renderCSVFileName()}
+        <Grid container justify='flex-start'>
+          <Box clone order={{ xs: 2, sm: 1 }}>
+            <Grid item xs={12} sm={9} className={rowLevelErrorClasses.table} >
+              <ValidationErrorTable invalidations={invalidations} />
+            </Grid>
+          </Box>
+          <Box clone order={{ xs: 1, sm: 2 }}>
+            <Grid item xs={12} sm={3} className={rowLevelErrorClasses.dialog}>
+              <Paper role='alert' >
+                <Typography>Review your CSV file</Typography>
+                <ErrorIcon className={rowLevelErrorClasses.dialogIcon} fontSize='large'/>
+                <Typography>Correct the file first and{renderUploadAgainButton()}</Typography>
```

Comment:
When the `No Schema error` happen the UI is presenting with error and upload again option as bullet list. I feel all 3 elements `Review Your CSV file` , 'Error message' and `upload again button` have some alignment anomalous. This could be formatted better. @janelilr any thoughts???

Reply:
This would be my suggestion: 
<img width="319" alt="Screen Shot 2021-05-06 at 8 57 36 AM" src="https://user-images.githubusercontent.com/46663379/117302375-8f013a00-ae49-11eb-92c0-08308b46058e.png">

Reply:
The exclamation point can stay. This is just a reference for aligning the text/text display.

Reply:
Everything is centered, but I can make the dialog smaller to make that look better.
@janelilr  Would it be better to give it a max width that is a fixed size (like 500px) or relative (like 75%)?

Reply:
75% 
![image](https://user-images.githubusercontent.com/17617998/117451409-7f98f400-af10-11eb-926e-7fee11739dd7.png)
![image](https://user-images.githubusercontent.com/17617998/117451440-8a538900-af10-11eb-8c10-a7cd994176c2.png)

500px
![image](https://user-images.githubusercontent.com/17617998/117451590-b1aa5600-af10-11eb-8458-9f3c16f12042.png)

![image](https://user-images.githubusercontent.com/17617998/117451577-ae16cf00-af10-11eb-914c-3c51655565b7.png)

Reply:
I set it to 75%